testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliControllerConfigTestCase.java

```diff
@@ -0,0 +1,438 @@
+     */
+    @Test
+    public void testImplicitSettings() {
+        JbossCliConfig jbossCliConfig = new JbossCliConfig(); //empty config
+        jbossCliConfig.writeJbossCliConfig(TMP_JBOSS_CLI_FILE);
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeNonInteractive();
+            String output = cli.getOutput();
+            assertConnected(output);
+        } catch (Exception ex) {
+            fail(ex.getLocalizedMessage());
+        } finally {
+            cli.destroyProcess();
+        }
+    }
+
+    /**
+     * Test to ensure https-remoting defaults to port 9993 when no port is specified
+     */
+    @Test
+    public void testHttpsRemotingConnection() {
+        Protocol expectedProtocol = HTTPS_REMOTING;
+        JbossCliConfig jbossCliConfig = new JbossCliConfig()
+                .setUseLegacyOverride(true).setDefaultProtocol(Protocol.REMOTE) //non-working default protocol
+                .setDefaultControllerProtocol(expectedProtocol)
+                .setDefaultControllerPort(null);
+        jbossCliConfig.writeJbossCliConfig(TMP_JBOSS_CLI_FILE);
+
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeNonInteractive();
+            String output = cli.getOutput();
+            assertTrue(output.contains(":" + 9993));
+            assertTrue(output.contains(expectedProtocol + "://"));
+            assertDisconnected(output);
+        } catch (Exception ex) {
+            fail(ex.getLocalizedMessage());
+        } finally {
+            cli.destroyProcess();
+        }
+    }
+
+    /**
+     * Returns CliProcessWrapper with settings loaded from TMP_JBOSS_CLI_FILE
+     *
+     * @param connectToAlias if true connects to aliased controller, {@code: <default-controller>} otherwise
+     * @return configured CliProcessWrapper
+     */
+    private CliProcessWrapper getTestCliProcessWrapper(boolean connectToAlias) {
+        CliProcessWrapper cli = new CliProcessWrapper()
+                .addCliArgument("-Djboss.cli.config=" + TMP_JBOSS_CLI_FILE.toPath())
+                .addCliArgument("--connect")
+                .addCliArgument("--echo-command")
+                .addCliArgument("--command=:read-attribute(name=server-state)");
+        if (connectToAlias) {
+            cli.addCliArgument("--controller=" + CONTROLLER_ALIAS_NAME);
+        }
+        return cli;
+    }
+
+    private void assertDisconnected(String output) {
+        assertTrue("connected when disconnected is expected", output.contains("Failed to connect to the controller"));
+        assertTrue("connected when disconnected is expected", output.contains("The connection failed"));
+        assertFalse("connected when disconnected is expected", output.contains("standalone@"));
+    }
+
+    private void assertConnected(String output) {
+        String expectedIP = "@" + TestSuiteEnvironment.getServerAddress() + ":" + TestSuiteEnvironment.getServerPort();
```

Comment:
We have some timing issue where the prompt would be not displayed when you ask for it. I would advise you to use the CLIProcessWrapper.pushLineAndWaitForResults(cmd, prompt). 
Instead of having: --command=:read-attribute(name=server-state)
You could call boolean ret = pushLineAndWaitForResults(":read-attribute(name=server-state)", null);
null prompt means the default connected prompt. ret will be true if the prompt is found. So should be safer to use that instead of assertConnected.